src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java

```diff
@@ -98,14 +98,19 @@ public void configure() throws Exception {
                     Map<String, Object> variables = exchange.getProperty(ZEEBE_VARIABLE, Map.class);
                     variables.put(PAYMENT_MODE, "gsma");
                     List<Transaction> transactionList = exchange.getProperty(TRANSACTION_LIST, List.class);
-                    while(transactionList.size() > 0) {
-                        GSMATransaction gsmaTransaction = convertTxnToGSMA(transactionList.get(0));
-                        exchange.setProperty(GSMA_CHANNEL_REQUEST, gsmaTransaction);
-                        exchange.setProperty(INIT_SUB_BATCH_FAILED, false);
-                        transactionList.remove(0);
-                    }
+                    exchange.setProperty("length",transactionList.size());
 
                 })
+                .loop(simple("${exchangeProperty.length}"))
+                .process(exchange -> {
+                    exchange.setProperty(LOOP_INDEX, exchange.getProperty("CamelLoopIndex", Integer.class));
```

Comment:
this is wrong.  LOOP INDEX already exist in Exchange. you can use that. 

to improve readability please assign this (`exchange.getProperty("CamelLoopIndex", Integer.class)`) in a local variable and use that while accessing the elements in the list